controllers/postController.js

```diff
@@ -7,27 +7,23 @@ const urlencodedParser = bodyParser.urlencoded({ extended: false });
 
 router.get(['/', '/posts'], (req, res) => {
   postService.getAllPosts().then((posts) => {
-    res.render('posts', { posts });
+    res.json(posts);
   }).catch((err) => {
     res.status(503).send(err.message);
   });
 });
 
 router.get('/posts/:post_id', (req, res) => {
   postService.getOnePost(req.params.post_id).then((post) => {
-    res.render('post_detail', { post });
+    res.json(post);
   }).catch((err) => {
     res.status(503).send(err.message);
   });
 });
 
-router.get('/post_new', (req, res) => {
-  res.render('post_new');
-});
-
 router.post('/post_new', urlencodedParser, (req, res) => {
```

Comment:
REST APIs typically don't have a dedicated url to create resources. This should be `/posts` instead of `/post_new`